api/health_test.go

```diff
@@ -30,11 +30,9 @@ func Test_HealthHandler_GetHealth(t *testing.T) {
 			validate: func(t *testing.T, resp *httptest.ResponseRecorder) {
```

Comment:
I think we should also add checks that we are setting the requestID as expected

You can define some request ID outside the for loop and then just add it to the context of the request
```Go
path := "/v1/health"
req, err := http.NewRequest(http.MethodGet, path, nil)
require.NoError(t, err)
resp := httptest.NewRecorder()

req = req.WithContext(requestIDWithContext(req.Context(), reqID))
```

Reply:
Thanks, I'll add that

Reply:
Addressed in https://github.com/hashicorp/consul-terraform-sync/pull/834/commits/9aea60d770d2b39bcc18b4495bd173cb5e46dfe1